### ISSUE TITLE
UnmarshalJSON wants to decode a ethermint/MsgEthereumTx but found a cosmos-sdk/StdTx

### ISSUE THREAD

**Issue by b1u3h4t** [cosmos/ethermint#90]
I'm developping on ethermint on branch: austin/genesisutil.
But I met a question: ERROR: UnmarshalJSON wants to decode a ethermint/MsgEthereumTx but found a cosmos-sdk/StdTx when I run "emintd gentx --name evm"

Below are detail steps:
```
emintd init moniker --chain-id 1
emintcli config chain-id 1 && emintcli config output json && emintcli config indent true && emintcli config trust-node true
emintcli emintkeys add evm
emintd add-genesis-account $(emintcli emintkeys show evm -a) 1000photon,100000000stake
emintd gentx --name evm
```
Then it produces an ERROR: UnmarshalJSON wants to decode a ethermint/MsgEthereumTx but found a cosmos-sdk/StdTx when I run "emintd gentx --name evm"

How can i fix it out, please help me, thanks!

**Comment by austinabell** [comment]
Yeah this is a known issue, just hasn't been documented. These modules aren't ready for use as it is waiting on changes within CosmosSDK. For now, you can just use:

```
emintcli keys add <name>
```

Instead of using `emintkeys` since I temporarily enabled cosmos keys to allow these genesis utils to be used to start the chain. (This should not work on that branch but will work off development)

I wouldn't develop off that branch, that is not the route that will be taken to get a functional chain running. I just kept it around in case we were to revisit creating ethermint specific genutils.

Let me know if I can explain anything else

**Comment by hussein1571** [comment]
Hello, 

After running emintd start  --pruning=nothing ...... The blockchain start normaly wiithout any problem.
But I don't understand till now, how can I use ethereum by sending transactions on ethermint.

Could you explain me please or send me commands that can I use after running this blockchain ? what can I do after?
**emintd start --pruning=nothing
I[2019-09-09|13:41:21.239] Starting ABCI with Tendermint                module=main 
I[2019-09-09|13:41:27.297] Executed block                               module=state height=3 validTxs=0 invalidTxs=0
I[2019-09-09|13:41:27.347] Committed state                              module=state height=3 txs=0 appHash=DB37E697D7ACD4ED9E94C71B3F1FC2E6169651A08ECBE07F0D75189ECB128CB1
I[2019-09-09|13:41:32.655] Executed block                               module=state height=4 validTxs=0 invalidTxs=0
I[2019-09-09|13:41:32.687] Committed state                              module=state height=4 txs=0 appHash=0BDE0E39D63DC44354138B39D5437DDAF873975908CCB4DDCCE7E6A8E889F5A6
I[2019-09-09|13:41:38.045] Executed block                               module=state height=5 validTxs=0 invalidTxs=0
I[2019-09-09|13:41:38.098] Committed state                              module=state height=5 txs=0 appHash=D4D1ABBD3E30F6F123A516E402A52C83DAC3B65F4AF64C9E02482BA9B2A8499D
I[2019-09-09|13:41:43.308] Executed block                               module=state height=6 validTxs=0 invalidTxs=0
I[2019-09-09|13:41:43.351] Committed state                              module=state height=6 txs=0 appHash=07E1917E7DC8FF2BC481D4AD50FEDCEF85D7B186F1BFFFE12552DC9D622B2084
I[2019-09-09|13:41:48.707] Executed block                               module=state height=7 validTxs=0 invalidTxs=0
I[2019-09-09|13:41:48.751] Committed state                              module=state height=7 txs=0 appHash=010C376BE8F51639B43EA9B1BD7DFD0D3167D7FB366D2F914671BCC5A244C6C4
I[2019-09-09|13:41:54.027] Executed block                               module=state height=8 validTxs=0 invalidTxs=0
I[2019-09-09|13:41:54.070] Committed state                              module=state height=8 txs=0 appHash=AFCE72E241273719A34015BABB2133783A42156DBF6F55EE79979469504C66A5
I[2019-09-09|13:41:59.380] Executed block                               module=state height=9 validTxs=0 invalidTxs=0
I[2019-09-09|13:41:59.434] Committed state                              module=state height=9 txs=0 appHash=30C85890DA2EC2DDEEEFFA5894389A01870678E27C76AB74A6A1F668CCD786**

**Comment by austinabell** [comment]
The tx processing is done but isn't connected to be able to be used on a running node yet, I have a WIP branch that would allow this hopefully very soon, but is not merged into development yet.

**Comment by b1u3h4t** [comment]
ok, good job, thanks so much, I will focus on it.

**Comment by hussein1571** [comment]
Hello @austinabell. when I use this command to create and sign a transaction

**emintcli tx send $(emintcli keys show name2 -a) 10photon --from=name1 --chain-id=testchain**

It produces this ERROR     
**ERROR: accepts 3 arg(s), received 2**

**Comment by austinabell** [comment]
> Hello @austinabell. when I use this command to create and sign a transaction
> 
> **emintcli tx send $(emintcli keys show name2 -a) 10photon --from=name1 --chain-id=testchain**
> 
> It produces this ERROR
> **ERROR: accepts 3 arg(s), received 2**

Hi @hussein1571,

All of this stuff is still wip and won't be very usable until a few more things are resolved, which may be another week.

Having said that, I haven't been working on the standard txs and haven't ensured they are functional as I have been focusing on the evm transactions. 

The problem with your command though is you aren't including a to address in the command. You can attach --help at the end of a command to get details about it with cobra, like:

```
emintcli tx send --help
```

But the evm state transition will be using `emintcli tx ethermint generate-eth-tx` and will be finalized when my wip branch comes into development.

**Comment by austinabell** [comment]
txs through cli and rpc available

**Comment by hussein1571** [comment]
@austinabell thank you man :+1:
